.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,16 @@
+### Description
+
+[Please describe what this change achieves]
+
+### Issues Resolved
+
+[List any existing issues this PR resolves, or any Discourse or
+StackOverflow discussion that's relevant]
+
+### Check List
+
+- [ ] New functionality includes tests
+- [ ] All tests pass
+- [ ] Documentation, especially RELEASE\_NOTES.md, has been updated if
```

Comment:
Probably mention `DOC_CHANGES.md` here because we need to be better about that for things below the level of release notes.